taskqueue/gcloud/aio/taskqueue/queue.py

```diff
@@ -44,25 +51,24 @@ def __init__(self, project: str, taskqueue: str,
 
     @backoff.on_exception(backoff.expo, Exception, max_tries=3)  # type: ignore
     async def _request(self, method: str, url: str,
-                       session: Optional[aiohttp.ClientSession] = None,
+                       session: Optional[Session] = None,
                        **kwargs: Any) -> Any:
         if not self.session:
-            self.session = aiohttp.ClientSession(conn_timeout=10,
-                                                 read_timeout=10)
-        s = session or self.session
+            self.session = RestSession()
+        s = RestSession(session) if session else self.session
         headers = await self.headers()
 
         resp = await s.request(method, url, headers=headers, **kwargs)
         # N.B. This is awaited early to give an extra helping hand to various
         # debug tools, which tend to be able to capture assigned variables but
         # not un-awaited data.
         data = await resp.json()
-        resp.raise_for_status()
```

Comment:
Why don't we want to `raise_for_status` anymore? When are we receiving exceptions? This seems like a change in behaviour of the library.

Reply:
https://github.com/talkiq/gcloud-aio/pull/101#discussion_r321904133

We're still calling `raise_for_status`, but its now called within the `resp = await s.request` call above.

Note that this breaks the `N.B.` comment above this line, though! Maybe we should emulate the `auto_raise_for_status` arg in requests here, so we can selectively not have the Session lib call this for us?

Reply:
Done!